mirrord-layer/src/pod_api.rs

```diff
@@ -107,126 +110,191 @@ pub async fn create_agent(config: LayerConfig) -> Result<Portforwarder> {
     )
     .await;
 
-    let agent_job_name = format!(
-        "mirrord-agent-{}",
-        Alphanumeric
-            .sample_string(&mut rand::thread_rng(), 10)
-            .to_lowercase()
-    );
-
     let agent_image = agent_image.unwrap_or_else(|| {
         concat!("ghcr.io/metalbear-co/mirrord:", env!("CARGO_PKG_VERSION")).to_string()
     });
 
-    let agent_pod: Job =
-        serde_json::from_value(json!({ // Only Jobs support self deletion after completion
-                "metadata": {
-                    "name": agent_job_name,
-                    "labels": {
-                        "app": "mirrord"
-                    }
-                },
-                "spec": {
-                "ttlSecondsAfterFinished": env_config.agent_ttl,
-
-                    "template": {
-                "spec": {
-                    "hostPID": true,
-                    "nodeName": runtime_data.node_name,
-                    "restartPolicy": "Never",
-                    "volumes": [
-                        {
-                            "name": "sockpath",
-                            "hostPath": {
-                                "path": runtime_data.socket_path
-                            }
-                        }
-                    ],
-                    "containers": [
-                        {
-                            "name": "mirrord-agent",
-                            "image": agent_image,
-                            "imagePullPolicy": image_pull_policy,
-                            "securityContext": {
-                                "privileged": true,
-                            },
-                            "volumeMounts": [
-                                {
-                                    "mountPath": runtime_data.socket_path,
-                                    "name": "sockpath"
-                                }
-                            ],
-                            "command": [
-                                "./mirrord-agent",
-                                "--container-id",
-                                runtime_data.container_id,
-                                "--container-runtime",
-                                runtime_data.container_runtime,
-                                "-t",
-                                "30",
-                            ],
-                            "env": [{"name": "RUST_LOG", "value": agent_rust_log}],
-                        }
-                    ]
+    let pod_name = if env_config.ephemeral_container {
```

Comment:
split the logic into two functions (ephemeral and old method) as this is getting too long